nca/Parsers/IstioPolicyYamlParser.py

```diff
@@ -485,30 +489,41 @@ def parse_ingress_rule(self, rule):
         to_array = self.get_key_array_and_validate_not_empty(rule, 'to')
         # currently parsing only ports
         # TODO: extend operations parsing to include other attributes
+        conn_props = ConnectivityProperties.make_empty_props()
         if to_array is not None:
             connections = ConnectionSet()
             for operation_dict in to_array:
-                connections |= self.parse_operation(operation_dict)
+                conns = self.parse_operation(operation_dict)
+                connections |= conns
+                conn_props |= conns.convert_to_connectivity_properties(self.peer_container)
```

Comment:
if the plan is to remove `ConnectionSet` after moving to the optimized reasoning, then this dependency in `conn_props |= conns.convert_to_connectivity_properties(self.peer_container)` should also be removed.

Reply:
Removed the dependency